broad/broad/settings.py

```diff
@@ -19,6 +19,12 @@
 
 SCHEDULER_PRIORITY_QUEUE = 'scrapy.pqueues.DownloaderAwarePriorityQueue'
 
+SCRAPY_BENCH_RANDOM_PAYLOAD_SIZE = None
+
+DOWNLOADER_MIDDLEWARES = {
```

Comment:
Providing this functionality creates an additional loop (scheduler -> middleware -> scheduler) and makes spider slower. We should use of [Spider Middleware](https://docs.scrapy.org/en/latest/topics/spider-middleware.html) here.

Reply:
Right, is the following method the one I need: https://docs.scrapy.org/en/latest/topics/spider-middleware.html#scrapy.spidermiddlewares.SpiderMiddleware.process_start_requests

I was a bit confused by the docs if the method only processes the `start_requests` property in a spider because of the name.